### DISCUSSION TITLE
Associate a color to a group of bookmarks

### DISCUSSION THREAD

**Discussion by aburlot** [organicmaps/organicmaps#1570] [Ideas]
I try to separate my bookmarks in different groups and I associated a colour with each group. But when I add a new bookmark to a group, I need to manually change the bookmark colour as the colour is associated with the bookmark and not the group. Would it be possible to associate a group and a colour?

**Comment by biodranik** [comment]
How would you design this feature, so it will be easy to use for everyone?

**Comment by aburlot** [comment]
Good question. I could imagine the group and the bookmark having both a colour attribute. The colour selection could be in the group option with the name edit. The bookmark could inherit its group colour when it is added in a group. That could be the default behaviour. Maybe an option could be added to allow the bookmark to override the group colour. If it's a mess, a function in the group option could force every bookmark to reset to the group colour.

The reason I suggested this is that I have a group for restaurant, a group for bars, a group for theaters, a group for people's places. I always have to remember which color is what after bookmarking something, and I have to check those bookmarks from time to time because I always make mistakes. I found redundant to have both groups and colours. How could I differentiate a blue bookmark in group 1 and a blue one in group 2? I cannot tell form the colour in wich group they are. If a colour is associated with a group by default, I would knwo they are not part of the same group.

Does that seem strange to you? My feeling of this might not be the same as everyone.
-------- Message d'origine --------

**Reply by biodranik** [reply]
The easiest way would be to always force the same color (or even symbol) for the whole group/set of bookmarks. But it will break the existing behavior. Forcing color in one case and not forcing in another looks as a complex UX, that's why I'm asking about the better approach.

**Reply by GhostInTheKayak** [reply]
Add an option to the ... menu for a bookmark list -- "Set colour".
( It is probably better to use "Set all icons" -- this allows for a future option of different icon shapes )
This displays a panel with all of the different POI icons available.
Selecting one resets all bookmarks in the list to use that icon.

The same UI would be valid in the "Edit bookmark" screen to set the icon for a single bookmark.

**Reply by biodranik** [reply]
Good idea! There is probably already an issue about it, can you please find and update it?

**Comment by erik55** [comment]
Great, did you know that you can just export your favorites from Osmand and import them directly in OrganicMaps!?? It worked. Just the colors of the groups were lost. That has to be done manually. But color import only works with kmz.

So:

* export favorites from Osmand (gpx files)
* import them in OrganicMaps
* export them from OrganicMaps **as kmz**
* unzip the files and edit the colors (I would do that with sed)
* zip them again and import into OrganicMaps those edited favorites-kmz files
* delete the first import in OrganicMaps that lacked the different colors

A small pull request could check if the imported gpx files are from osmand and then apply the corresponding colors.

**Reply by cyber-toad** [reply]
@erik55 it looks like some osmand-specific color encoding is not supported in OM. Could you share a sample file with such group colors? For privacy you can set any random names/coordinates for labels, I just need to see what tags/attributes are used to store colors.
